src/source/JavaGenerator.scala

```diff
@@ -534,11 +534,11 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
         case MOptional => {
           if (inOptional)
           	throw new AssertionError("nested optional?")
-          w.wl(s"if (this.${idJava.field(f.ident)} != null) {").nested {
+          w.wl(s"if (this.${idJava.field(f.ident)} != null)").braced {
```

Comment:
I think the old solution here was better because it generates `if (...) {` and `} else {` on single lines.  The current output is messily long.  Can you either revert to this behavior in both parts of this generator, or arrange to get the shorter output with braced?